core/src/test/java/com/google/errorprone/refaster/UFreeIdentTest.java

```diff
@@ -41,10 +49,36 @@ public void inlinesExpression() {
 
   @Test
   public void binds() {
-    JCExpression expr = parseExpression("\"abcdefg\".charAt(x + 1)");
-    UFreeIdent ident = UFreeIdent.create("foo");
-    assertThat(ident.unify(expr, unifier)).isNotNull();
-    assertThat(unifier.getBindings()).containsExactly(new UFreeIdent.Key("foo"), expr);
+    CompilationTestHelper.newInstance(UnificationChecker.class, getClass())
+        .addSourceLines(
+            "A.java",
+            "class A {",
+            "  public void bar() {",
+            "    int x = 0;",
+            "    \"abcdefg\".charAt(x + 1);",
+            "  }",
+            "}")
+        .doTest();
+  }
+
+  @BugPattern(
+      name = "UnificationChecker",
+      summary = "Verify that unifying the expression results in the correct binding",
+      explanation = "For test purposes only",
+      severity = SUGGESTION)
+  public static class UnificationChecker extends BugChecker
+      implements ExpressionStatementTreeMatcher {
+    @Override
+    public Description matchExpressionStatement(ExpressionStatementTree tree, VisitorState state) {
+      Unifier unifier = new Unifier(new Context());
+      UFreeIdent ident = UFreeIdent.create("foo");
+
+      assertThat(ident.unify(tree.getExpression(), unifier)).isNotNull();
```

Comment:
We can avoid the `.getExpression()` calls if we use a `MethodInvocationTreeMatcher` instead.